src/mpl2/test/cpp/TestSnapper.cpp

```diff
@@ -31,42 +32,254 @@ class Mpl2SnapperTest : public ::testing::Test
+                                             "simple_master",
+                                             1000,
+                                             1000,
+                                             odb::dbMasterType::BLOCK,
+                                             db_->getTech()->findLayer("L1"));
+
+  odb::dbInst* inst = odb::dbInst::create(block, master, "macro1");
+
+  Snapper snapper;
+  snapper.setMacro(inst);
+
+  // Considering the grid pattern configuration (0 20 40 ... 980)
+  // manufacturing grid size (5), and direction preference (horizontal):
+  // Valid alignments for x would include 10 15 20 25 30 35 40 ...
+  // with 14 to 11 snapping to 10, 19 to 16 snapping to 15, etc.
+  // Valid alignments for y would include 15 35 55 75 95 ...
+  // with 40 to 59 snapping to 15, 60 to 79 mapping to 35, etc.
+
+  inst->setOrigin(511, 540);
+  snapper.snapMacro();
+  EXPECT_TRUE(inst->getOrigin().x() == 510);
+  EXPECT_TRUE(inst->getOrigin().y() == 515);
+
+  inst->setOrigin(514, 559);
+  snapper.snapMacro();
+  EXPECT_TRUE(inst->getOrigin().x() == 510);
+  EXPECT_TRUE(inst->getOrigin().y() == 515);
+
+  inst->setOrigin(516, 560);
+  snapper.snapMacro();
+  EXPECT_TRUE(inst->getOrigin().x() == 515);
+  EXPECT_TRUE(inst->getOrigin().y() == 535);
+
+  inst->setOrigin(519, 579);
+  snapper.snapMacro();
+  EXPECT_TRUE(inst->getOrigin().x() == 515);
+  EXPECT_TRUE(inst->getOrigin().y() == 535);
+
+}  // SnapMacrosHorizontal
+
+TEST_F(Mpl2SnapperTest, SnapMacrosVertical)
+{
+  // This test checks snapMacro functionality
+  // for a block with 1 master (vertical direction layer preference)
+
+  utl::Logger* logger = new utl::Logger();
+  odb::dbDatabase* db_ = createSimpleDB();
+  db_->setLogger(logger);
+
+  odb::dbBlock* block = odb::dbBlock::create(db_->getChip(), "simple_block");
+  block->setDieArea(odb::Rect(0, 0, 1000, 1000));
+
+  createSimpleTrack(block, db_->getTech()->findLayer("L1"), 0, 50, 20, 5);
+
+  db_->getTech()->findLayer("L1")->setDirection(odb::dbTechLayerDir::VERTICAL);
+
+  odb::dbMaster* master = createSimpleMaster(db_->findLib("lib"),
+                                             "simple_master",
+                                             1000,
+                                             1000,
+                                             odb::dbMasterType::BLOCK,
+                                             db_->getTech()->findLayer("L1"));
+
+  odb::dbInst* inst = odb::dbInst::create(block, master, "macro1");
+
+  Snapper snapper;
+  snapper.setMacro(inst);
+
+  // Considering the grid pattern configuration (0 20 40 ... 980)
+  // manufacturing grid size (5), and direction preference (horizontal):
+  // Valid alignments for x would include 15 35 55 75 95 ...
+  // with 40 to 59 snapping to 15, 60 to 79 mapping to 35, etc.
+  // Valid alignments for y would include 10 15 20 25 30 35 40 ...
+  // with 14 to 11 snapping to 10, 19 to 16 snapping to 15, etc.
```

Comment:
We should be able to see clearly the pins dimensions from the scope of the test in order to avoid having to read a long comment such as this one in order to figure it out.

Reply:
I've re-organized the helper functions so that the MPin instantiations are now also "visible", does this work?